packages/core/svelte/src/walletStore.ts

```diff
@@ -103,35 +143,32 @@ export async function initialize({
     localStorageKey: string;
     onError?: ErrorHandler;
 }): Promise<void> {
-    walletConfigStore.update((storeValues: WalletConfigStore) => ({
-        ...storeValues,
+    walletConfigStore.set({
         wallets,
         walletsByName: wallets.reduce((walletsByName, wallet) => {
             walletsByName[wallet.name] = wallet;
             return walletsByName;
         }, {} as WalletDictionary),
         autoConnect,
         localStorageKey,
-        onError,
-    }));
+        onError
+    });
 
-    walletNameStore.update((storeValues: WalletNameStore) => ({
-        ...storeValues,
-        walletName: getLocalStorage<WalletName>(localStorageKey),
-    }));
+    const walletName = getLocalStorage<WalletName>(localStorageKey);
+
+    if (walletName) {
```

Comment:
Prevents setting walletName to `null` if local storage is empty. Not having this check doesn't have any adverse effects, just prevents unnecessary side effects from running